staging/src/k8s.io/cli-runtime/pkg/resource/builder.go

```diff
@@ -803,7 +803,7 @@ func (b *Builder) mappingFor(resourceOrKindArg string) (*meta.RESTMapping, error
 		// if the error is _not_ a *meta.NoKindMatchError, then we had trouble doing discovery,
 		// so we should return the original error since it may help a user diagnose what is actually wrong
 		if meta.IsNoMatchError(err) {
-			return nil, fmt.Errorf("the server doesn't have a resource type %q", groupResource.Resource)
+			return nil, fmt.Errorf("the server doesn't have a resource type %q: %w", groupResource.Resource, err)
```

Comment:
After investigating this carefully, I think we can't change this due to backwards compatibility reasons without breaking anywhere. Because it is one of the functions that widely used and probably some scripts in somewhere rely on this error message which is now we are changing.

Reply:
There have been cases for similar errors where the handling was adopted: https://github.com/kubernetes/kubernetes/commit/dfcc141c91925c5cc7b04ed50bb9d51aa1b8b343

But I do agree changing error messages should be done with care. 
In this case I hoped it would be no issue, since the message is only appended to.